### PR TITLE
Fix building CoreCLR with Clang 3.9

### PR DESCRIPTION
**Pull request by janvorli**
There were few constructs that Clang 3.9 didn't like due to its strict
C++ standard conformance rules.

### CODE REVIEW

**Comment by janvorli** [comment]
@jkotas can you take a look please?

File: src/debug/daccess/dacdbiimpl.cpp [hunk 1]
```diff
@@ -90,7 +90,6 @@ IDacDbiInterface::IAllocator * g_pAllocator = NULL;
 //
 
 // Need a class to serve as a tag that we can use to overload New/Delete.
```
Comment (jkotas) [inline src/debug/daccess/dacdbiimpl.cpp]: Nit - delete this comment?

**Comment by jkotas** [comment]
ARM builds are failing with `undefined symbol: interopsafeEXEC`

**Comment by jkotas** [comment]
LGTM otherwise

**Comment by janvorli** [comment]
@jkotas SELECTANY doesn't work as expected on ARM for some reason. I have replaced that by instances in related .cpp files.

**Pull request merged**